libsolidity/parsing/Parser.cpp

```diff
@@ -1313,11 +1314,32 @@ ASTPointer<InlineAssembly> Parser::parseInlineAssembly(ASTPointer<ASTString> con
 
 	expectToken(Token::Assembly);
 	yul::Dialect const& dialect = yul::EVMDialect::strictAssemblyForEVM(m_evmVersion);
+	ASTPointer<vector<ASTPointer<ASTString>>> dialectFlags;
 	if (m_scanner->currentToken() == Token::StringLiteral)
 	{
-		if (m_scanner->currentLiteral() != "evmasm")
-			fatalParserError(4531_error, "Only \"evmasm\" supported.");
+		std::string const& dialectString = m_scanner->currentLiteral();
+		auto it = dialectString.begin();
+		auto end = dialectString.end();
+		auto dialectStart = it;
+		while (it != end && !isWhiteSpace(*it))
+			++it;
+
 		// This can be used in the future to set the dialect.
+		if (string(dialectStart, it) != "evmasm")
```

Comment:
Do we really require 'evmasm' now? Can't it just be the default?

Reply:
You mean ``assembly "memory-safe" {`` would be fine?
I'm not sure... the idea so far was to require the first world to be the dialect and then allow flags later than that... having the dialect itself just be a flag at arbitrary position... would probably also work, though.

Reply:
Do you like this syntax in general? I'm not sure ``assembly "evmasm memory-safe" {`` really looks good - and there's little reason for not making it ``assembly "evmasm" "memory-safe" {}``, is there? Then we could also distinguish more explicitly between dialect and flags, e.g. as ``assembly "evmasm" ("memory-safe") {}``... and then allow omitting the dialect defaulting to ``"evmasm"``... Or something like that... generally what I mean to say is: I'm not sure about the syntax yet :-).

Reply:
Yeah, I think this requires further discussion.